src/diffusers/loaders.py

```diff
@@ -968,20 +998,24 @@ def text_encoder_lora_attn_procs(self):
             return self._text_encoder_lora_attn_procs
         return
 
-    def _remove_text_encoder_monkey_patch(self):
-        # Loop over the CLIPAttention module of text_encoder
-        for name, attn_module in self.text_encoder.named_modules():
-            if name.endswith(TEXT_ENCODER_ATTN_MODULE):
-                # Loop over the LoRA layers
-                for _, text_encoder_attr in self._lora_attn_processor_attr_to_text_encoder_attr.items():
-                    # Retrieve the q/k/v/out projection of CLIPAttention
-                    module = attn_module.get_submodule(text_encoder_attr)
-                    if hasattr(module, "old_forward"):
-                        # restore original `forward` to remove monkey-patch
-                        module.forward = module.old_forward
-                        delattr(module, "old_forward")
-
-    def _modify_text_encoder(self, attn_processors: Dict[str, LoRAAttnProcessor]):
+    @classmethod
+    def _remove_text_encoder_monkey_patch(cls, text_encoder):
+        for _, attn_module in text_encoder_attn_modules(text_encoder):
+            if isinstance(text_encoder, CLIPTextModel):
+                attn_module.q_proj = attn_module.q_proj.regular_linear_layer
+                attn_module.k_proj = attn_module.k_proj.regular_linear_layer
+                attn_module.v_proj = attn_module.v_proj.regular_linear_layer
+                attn_module.out_proj = attn_module.out_proj.regular_linear_layer
+            elif isinstance(text_encoder, T5EncoderModel):
+                attn_module.q = attn_module.q.regular_linear_layer
+                attn_module.k = attn_module.k.regular_linear_layer
+                attn_module.v = attn_module.v.regular_linear_layer
+                attn_module.o = attn_module.o.regular_linear_layer
+            else:
+                raise ValueError(f"{text_encoder.__class__.__name__} does not support LoRA training")
+
+    @classmethod
+    def _modify_text_encoder(cls, attn_processors: Dict[str, LoRAAttnProcessor], text_encoder, lora_scale=1):
```

Comment:
We can move this to a class method to avoid instantiating a copy of the pipeline to run the function and just explicitly pass the two additional necessary arguments in.

Ideally this doesn't even have to be defined on the mixin and could be defined on the module but I'm trying to keep the diff more readable